modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/servlet/taglib/DeferredScriptsBottomDynamicInclude.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.web.internal.servlet.taglib;
+
+import com.liferay.frontend.js.web.internal.script.DeferredScriptsImpl;
+import com.liferay.frontend.js.web.internal.script.DeferredScriptsManagerImpl;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = DynamicInclude.class)
+public class DeferredScriptsBottomDynamicInclude implements DynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		DeferredScriptsImpl deferredScriptsImpl =
+			_deferredScriptsManagerImpl.getDeferredScripts();
+
+		if (deferredScriptsImpl != null) {
+			PrintWriter printWriter = httpServletResponse.getWriter();
+
+			for (String script : deferredScriptsImpl.getScripts()) {
+				printWriter.print("<script type=\"module\">");
+				printWriter.print(script);
+				printWriter.print("</script>");
+			}
+		}
+	}
+
+	@Override
+	public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
+		dynamicIncludeRegistry.register("/html/common/themes/bottom.jsp#post");
```

Comment:
I noticed in one of the files above that we have `position="inline"` for a script tag and then here we are registering to the `bottom.jsp`. Will that be an issue or cause conflicts?

Reply:
Mmm, that's incorrect most probably. It may not cause issues, but if it was inline, it should be printed directly instead of registered with DeferredScriptsManager... Where was that?

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/2153/files#diff-8c8cdc08aac294bbb2bfbb8201d5c2568e4a25ec8aad85abe3460e870f1f5d8cR212

Reply:
I see...

But we are only using `DeferredScripts` from [`ReactRendererUtil`](https://github.com/liferay-frontend/liferay-portal/pull/2153/files#diff-de41dd587f07cdf94ec439e7dd79436cda7646c89194bf319bc1c3ca7b2e7b23R229), which is only intended to be used from `ReactRenderer` directly or from the `<react:component>` tag.

Given that `ReactRenderer` controls inlining using the [`ComponentDescriptor` class](https://github.com/liferay/liferay-portal/blob/0c9f52de7a10307fb825fbdf0f371aba4cc247a1/modules/apps/portal-template/portal-template-react-renderer-api/src/main/java/com/liferay/portal/template/react/renderer/ComponentDescriptor.java#L78) and we honor that [in `ReactRendererUtil`](https://github.com/liferay-frontend/liferay-portal/pull/2153/files#diff-de41dd587f07cdf94ec439e7dd79436cda7646c89194bf319bc1c3ca7b2e7b23R229), we are safe.